slides/week6/week6.org

```diff
@@ -0,0 +1,57 @@
+#+TITLE: Week 6
+#+AUTHOR: Sahit Chintalapudi 
+#+EMAIL: schintalapudi@gatech.edu
+
+* Computer Vision
+- We have the knowledge in C++ to describe the logic we might want a robot to
+  have. But we need to be able to make sense of what the robot sees and
+  classify it before we can act on this logic.
+- Cue OpenCV, an open source computer vision library with bindings for C++
+  (and a few other languages)
+- I guess our ability to see has been ++'d
+
+* What does an Image look like to your computer?
+- OpenCV stores images in an object called a /Mat/
+- A Mat is matrix with rows and columns! In a grayscale image. each element
+  of the Mat is a pixel in the image and its location in the Mat corresponds 
+  to its location in the image
+- If the image is grayscale, then a pixel is an int from 0 (black) to 255
+  (white)
+- Computers have no concept of "2d", so Images in memory are /continuous/.
+  This means each row of the image is appended onto the end of the last. To
+  iterate through a Mat you just get a pointer to the beginning of the first
+  row and keep track of your row number by how far you've traversed.
+
+* Color Images
+- Color images don't embed the color of a pixel in one element. Often, you'll
+  find each pixel represented in BGR (Blue component, Green Component, Red
+  Component) form. So now, each row of a color image is 3 times as long as a
+  row of a black and white image.
+[[https://docs.opencv.org/2.4/_images/math/b6df115410caafea291ceb011f19cc4a19ae6c2c.png]]
+- Images don't have to be stored in just BGR format!
+
+* HSV Images
+- Each Pixel in a color image has a hue, a saturation, and a luminosity. 
+- The nice thing about looking at colors from this perspective is that we're
+  more tolerant to changes in lighting conditions making the same color look
+  different.
+[[https://image.slidesharecdn.com/01presentationhuehistograms-150707215651-lva1-app6892/95/about-perception-and-hue-histograms-in-hsv-space-5-638.jpg]]
+
+* Finding the blue in an image
+#+BEGIN_SRC c++
+Mat findBlue(const Mat& frameBGR) { 
```

Comment:
Ah, that makes a lot more sense now, thanks for the follow up. I have a few more questions though:

* The low value that you're picking `HSV(78, 50, 70)` seems 'green' to me, [is that normal](https://duckduckgo.com/?q=HSV(78%2C50%2C70)&kc=1&k8=a6aaab&k1=1&kx=909396&ky=2e2e2e&kaa=B68800&kp=1&k9=ce4139&kj=262626&kam=osm&k5=1&kaj=m&kn=-1&k7=262626&kao=1&kae=d&kv=1&kak=-1&ia=answer)
* Can you link some source on what a CVT is? I'm only finding a super fancy transmission for cars when I search or do an acronym lookup.

I'm fine with leaving the comments and notes out personally, since the comments are in the code.